sei-tendermint/types/node_info_test.go

```diff
@@ -178,76 +180,74 @@ func TestNodeInfoAddChannel(t *testing.T) {
 	require.Contains(t, nodeInfo.Channels, byte(0x02))
 }
 
+func TestResolveAddressStringDNS(t *testing.T) {
+	addr, err := ResolveAddressString("deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@localhost:1234")
+	require.NoError(t, err)
+	require.True(t, addr.Addr().IsLoopback())
+	require.Equal(t, 1234, addr.Port())
+}
+
 func TestResolveAddressString(t *testing.T) {
 	testCases := []struct {
-		name     string
-		addr     string
-		expected string
-		correct  bool
+		name    string
+		addr    string
+		want    netip.AddrPort
+		correct bool
```

Comment:
A small convention I follow when writing go tests "want" and "got" (as you have used) but the name qualified for both result and error:

```suggestion
		wantAddrPort    netip.AddrPort
		wantErr           bool
```

Better yet, change `wantErr` to string and infer empty as "want no err", and non-empty as `require.ErrorContains` for specific type of error.

Reply:
I strongly prefer strongly typed error to "stringly" typed error.